packages/astro/src/runtime/server/index.ts

```diff
@@ -300,7 +300,7 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 		if (metadata.hydrate === 'only') {
 			html = await renderSlot(result, slots?.fallback);
 		} else {
-			({ html } = await renderer.ssr.renderToStaticMarkup(Component, props, children, metadata));
+			({ html } = await renderer.ssr.renderToStaticMarkup.call({ result }, Component, props, children, metadata));
```

Comment:
Exposes `result` to renderers so they can access them directly, might be necessary for some complex custom integrations.

Reply:
Is there a specific use-case you have in mind?

Reply:
Currently using it for MDX support, but it might not be necessary for long given your work on removing script/style hoisting.

OK to leave as is for now, with the full knowledge that it's likely going away very soon?

Reply:
What does MDX specifically need the result for? To push in scripts/styles? If that's the case I don't think we should add a feature that blocks streaming. Instead include a `<script>`/`<style>` tag as part of the returned HTML.